### PR TITLE
GH10: Implementing Support for .NET Core

### PR DESCRIPTION
**Pull request by AdmiringWorm**
The code have thus far not been tested on Linux and OSX (or travis to be exact).
Currently it's expected to fail on Linux/OSX during either build, or unit testing.

### CODE REVIEW

File: Cake.Recipe/Content/nuget.cake [hunk 1]
```diff
@@ -1,3 +1,34 @@
...
+        ArgumentCustomization = (args) => {
+            if (BuildParameters.ShouldBuildDotNetCoreSourcePackage)
+            {
+                args.Append("--include-source");
+            }
+            return args
+                .Append("/p:Version={0}", BuildParameters.Version.SemVersion)
+                .Append("/p:AssemblyVersion={0}", BuildParameters.Version.Version)
+                .Append("/p:FileVersion={0}", BuildParameters.Version.Version)
+                .Append("/p:AssemblyInformationalVersion={0}");
```
Comment (gep13) [inline Cake.Recipe/Content/nuget.cake]: @AdmiringWorm I think you are missing `BuildParameters.Version.InformationalVersion` here?

File: Cake.Recipe/Content/testing.cake [hunk 2]
```diff
@@ -136,12 +134,56 @@ BuildParameters.Tasks.TestFixieTask = Task("Test-Fixie")
...
-{
+BuildParameters.Tasks.DotNetCoreTestTask = Task("DotNetCore-Test")
+    .IsDependentOn("Install-OpenCover")
+    .Does(() => {
+    // Since we need to run dotnet test for each unit test library
+    // and merge the coverage result at each run, we need to delete
+    // the existing file in case it exists from a previous run.
+    if (FileExists(BuildParameters.Paths.Files.TestCoverageOutputFilePath))
+    {
+        DeleteFile(BuildParameters.Paths.Files.TestCoverageOutputFilePath);
```
Comment (gep13) [inline Cake.Recipe/Content/testing.cake]: @AdmiringWorm isn't this covered by the main `Clean` task?

File: Cake.Recipe/Content/nuget.cake (hunk 1 above)
Comment (AdmiringWorm) [inline Cake.Recipe/Content/nuget.cake]: Ops, yeah you're right, how silly of me.

File: Cake.Recipe/Content/testing.cake (hunk 2 above)
Comment (AdmiringWorm) [inline Cake.Recipe/Content/testing.cake]: oh right, completely forgot.
Based it on my own build script for Cake.Transifex, which doesn't clean the coverage file before the unit testing task.

**Comment by gep13** [comment]
@AdmiringWorm is this marked as WIP due to the lack of testing on LInux and OSX?  If so, then I would be happy to YOLO this just now, and carry out some further testing with re-work if required.  I know for a fact that Cake.Recipe isn't working on Travis just now due to another issue, so the changes that you have here won't work until that is corrected.  Happy to have .NET core support on work on AppVeyor just now, and then extent out from there.

Thoughts?

**Comment by AdmiringWorm** [comment]
Yes, it's mainly because of lack of travis testing.
But also because I have a feeling I'm missing something, but can't for the life of me think of what.
I do now it currently works on appveyor though, as can be seen here: https://ci.appveyor.com/project/cakecontrib/cake-codecov/build/0.2.1-using-cake-recipe.1+19.build.26

So I don't mind it being merged, I'll remove the `WIP` for now though.

**Comment by gep13** [comment]
@AdmiringWorm ok, cool.  I will take this for a proper spin tomorrow lunch time on the Cake.HockeyApp repo, which is where I started doing the conversion to support .NET core.

**Comment by AdmiringWorm** [comment]
Sounds good, but before you do. I'll be renaming one of the added arguments to a more generically named one so it can be reused.

I added a argument to compile source nupkg packages using dotnet pack, I'm thinking to perhaps rename that variable and use it to decide whether to generate source nupkg packages through the non-dotnet task as well.

**Comment by gep13** [comment]
@AdmiringWorm how does that work with GitLink which is currently being used in Cake.Recipe.  This allows for linking the source files directly to the GitHub repository where it is hosted, therefore you don't need to include the source files within the package.

**Comment by AdmiringWorm** [comment]
Honestly, I don't know. I've never used GitLink.
But I'm thinking of it as an alternative for those that don't use GitLink, but are using the SymbolSource server instead.

**Comment by gep13** [comment]
@AdmiringWorm yip, I would have no objection to that 👍

**Comment by AdmiringWorm** [comment]
Also, it won't include the source inside the normal package, but rather a separate package that will be uploaded to symbolsource.

**Comment by AdmiringWorm** [comment]
Of course, would need to add some code to the task that publishes nupkg packages to nuget.org so it ignores packages ending with `symbols.nupkg` (I think that was the name), as they will automatically be pushed to the correct destination anyhow.

Not sure how to handle MyGet though.

**Comment by gep13** [comment]
So, only thing I would add is that "I think" GitLink always runs currently, so we should probably switch to make that fully configurable, allowing the user to control what is run.

**Comment by gep13** [comment]
>@AdmiringWorm said...
Not sure how to handle MyGet though.

I "think" this just works, but I would have to check, been a long time since I have done anything with that.

**Comment by AdmiringWorm** [comment]
> So, only thing I would add is that "I think" GitLink always runs currently

You do have the option to turn of GitLink though, by setting `shouldExecuteGitLink` to `false`

Anyways, these are changes that should be done in a separate PR IMO, so for now I'm just renaming the parameter.

**Comment by gep13** [comment]
>@AdmiringWorm said...
Anyways, these are changes that should be done in a separate PR IMO, so for now I'm just renaming the parameter.

Agreed.

**Comment by AdmiringWorm** [comment]
@gep13 I opened #112 in preparation to adding the logic necessary for creating symbol/source packages.

**Comment by AdmiringWorm** [comment]
@gep13 any reason why the wyam task would fail after this PR was merged?

**Comment by gep13** [comment]
This is a known issue that @pascalberger noticed, and which I thought I had fixed.  Was actually just speaking to @devlead about it, as it is linked to his Cake.Git addin.

Nothing for you to worry about in this PR, but something that may need to be re-investigated.

**Comment by gep13** [comment]
This was the issue in question...

https://github.com/cake-contrib/Cake.Recipe/issues/104

**Pull request merged**